mapo/agents/mapo/off_mapo_policy.py

```diff
@@ -302,26 +329,24 @@ class TargetUpdatesMixin:  # pylint: disable=too-few-public-methods
     def build_update_targets_op(self, tau=None):
         """Build op to update target networks."""
         tau = tau or self.config["tau"]
-        main_variables = self.model.variables()
-        target_variables = self.target_model.variables()
         update_target_expr = [
-            target_var.assign(tau * var + (1.0 - tau) * target_var)
-            for var, target_var in zip(main_variables, target_variables)
+            target.assign(tau * main + (1.0 - tau) * target)
+            for main, target in self.model.main_and_target_variables
         ]
         return tf.group(*update_target_expr)
 
 
 OffMAPOTFPolicy = MAPOTFPolicy.with_updates(  # pylint: disable=invalid-name
```

Comment:
What does `OffMAPOTFPolicy` depend / extend on `MAPOTFPolicy` ?

Reply:
They share two methods, which are 
```python
    optimizer_fn=create_separate_optimizers,
    gradients_fn=compute_separate_gradients,
```
I can copy the two from `MAPOTFPolicy` and declare `OffMAPOTFPolicy` with `build_tf_policy` to avoid confusion.

Reply:
I personally would prefer to avoid implicit dependencies like this in the future... but no worries